File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size()-1;
        int ans = INT_MAX;

        while(r>=l)
        {
            
            int mid = (l+r)/2;
            cout<<nums[mid]<<endl;
            ans = min(ans,nums[mid]);
            if(nums[l]<nums[mid] && nums[mid]<nums[r])
            {
                return nums[l];
            }
            if(nums[mid]>=nums[l])
            {
                l=mid+1;
            }
            else
            {
                r=mid-1;
            }

        }

        return ans;

        
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-3.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        # kth largest -> we need to fetch a smaller element from heap head
        # so we use a min heap
        h = []
        heapq.heapify(h)
        for n in nums:
            if len(h) < k:
                heapq.heappush(h, n)
                continue
            if n > h[0]:
                heapq.heappop(h)
                heapq.heappush(h,n)
        return h[0]
            
            
        
        